source/game/field/RKGndCol.cc

```diff
@@ -0,0 +1,374 @@
+#include "RKGndCol.hh"
+
+#include <cmath>
+
+#include "egg/math/Math.hh"
+
+#include "game/field/KCollisionTypes.hh"
+
+// Some functions from m8 -
+// https://github.com/em-eight/mkw/tree/master/source/game/field/RKGndCol.cpp
+
+namespace Field {
+
+RKGndCol::RKGndCol() = default;
+
+RKGndCol::RKGndCol(const void *file) {
+    auto addOffset = [](const void *file, u32 offset) -> const void * {
+        return reinterpret_cast<const void *>(reinterpret_cast<const u8 *>(file) + offset);
+    };
+    u8 *unsafeData = reinterpret_cast<u8 *>(const_cast<void *>(file));
+    EGG::RamStream stream = EGG::RamStream(unsafeData, sizeof(KColHeader));
+
+    u32 posOffset = stream.read_u32();
+    u32 nrmOffset = stream.read_u32();
+    u32 prismOffset = stream.read_u32();
+    u32 blockOffset = stream.read_u32();
+
+    assert(posOffset == sizeof(KColHeader));
```

Comment:
This is an unsafe assert.

Reply:
Can you clarify what you mean? This assert is an indication that we are reading from a malformed source file, no?

Reply:
It's possible for the header to be variable size (in fact, I think more MKW modders should do this), but we also cannot guarantee that the position section will come after the header, as they are entirely interchangeable.